### ISSUE TITLE
Packer crash with Proxmox builder

### ISSUE THREAD

**Issue by AlekKCode** [hashicorp/packer-plugin-proxmox#185]
#### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

#### Overview of the Issue

packer build -var-file='../credentials.pkr.hcl' ubuntu-server-focal.pkr.hcl
When trying to build template, packer crashes with error: 

Build 'ubuntu-server-focal.proxmox.ubuntu-server-focal' errored after 50 milliseconds 233 microseconds: unexpected EOF

panic: runtime error: invalid memory address or nil pointer dereference
2023/04/10 22:13:34 packer-builder-proxmox plugin: [signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x301f9d5]

#### Reproduction Steps

Ubuntu desktop 22.04.2 amd64
I tried packer 1.8.6 and 1.8.0, both crashed

### Simplified Packer Template

essentially the same template as https://github.com/ChristianLempa/boilerplates/tree/main/packer/proxmox/ubuntu-server-focal

### Operating system and Environment details

Ubuntu 22.4.2 desktop amd64
### Log Fragments and crash.log files

2023/04/10 22:39:49 [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/04/10 22:39:49 [TRACE] discovering plugins in /usr/local/bin
2023/04/10 22:39:49 [TRACE] discovering plugins in /root/.config/packer/plugins
2023/04/10 22:39:49 [TRACE] discovering plugins in .
2023/04/10 22:39:49 [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/04/10 22:39:49 [INFO] PACKER_CONFIG env var set; attempting to open config file: /root/.packerconfig
2023/04/10 22:39:49 [WARN] Config file doesn't exist: /root/.packerconfig
2023/04/10 22:39:49 [INFO] Setting cache directory: /root/.cache/packer
2023/04/10 22:39:49 [TRACE] validateValue: not active for proxmox_api_token_secret, so skipping
2023/04/10 22:39:49 [TRACE] validateValue: not active for proxmox_api_url, so skipping
2023/04/10 22:39:49 [TRACE] validateValue: not active for proxmox_api_token_id, so skipping
2023/04/10 22:39:49 [TRACE] Starting internal plugin packer-builder-proxmox
2023/04/10 22:39:49 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-builder-proxmox"}
2023/04/10 22:39:49 Waiting for RPC address for: /usr/local/bin/packer
2023/04/10 22:39:49 packer-builder-proxmox plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/04/10 22:39:49 packer-builder-proxmox plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/04/10 22:39:49 packer-builder-proxmox plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /root/.packerconfig
2023/04/10 22:39:49 packer-builder-proxmox plugin: [WARN] Config file doesn't exist: /root/.packerconfig
2023/04/10 22:39:49 packer-builder-proxmox plugin: [INFO] Setting cache directory: /root/.cache/packer
2023/04/10 22:39:49 packer-builder-proxmox plugin: args: []string{"packer-builder-proxmox"}
2023/04/10 22:39:49 Received unix RPC address for /usr/local/bin/packer: addr is /tmp/packer-plugin2663617777
2023/04/10 22:39:49 packer-builder-proxmox plugin: Plugin address: unix /tmp/packer-plugin2663617777
2023/04/10 22:39:49 packer-builder-proxmox plugin: Waiting for connection...
2023/04/10 22:39:49 packer-builder-proxmox plugin: Serving a plugin connection...
2023/04/10 22:39:49 packer-builder-proxmox plugin: Number of sockets 0 is too small, using default: 1
2023/04/10 22:39:49 packer-builder-proxmox plugin: CPU type not set, using default 'kvm64'
2023/04/10 22:39:49 packer-builder-proxmox plugin: OS not set, using default 'other'
2023/04/10 22:39:49 packer-builder-proxmox plugin: Disk 0 cache mode not set, using default 'none'
2023/04/10 22:39:49 [TRACE] Starting internal plugin packer-provisioner-shell
2023/04/10 22:39:49 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-provisioner-shell"}
2023/04/10 22:39:49 Waiting for RPC address for: /usr/local/bin/packer
2023/04/10 22:39:49 packer-provisioner-shell plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/04/10 22:39:49 packer-provisioner-shell plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/04/10 22:39:49 packer-provisioner-shell plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /root/.packerconfig
2023/04/10 22:39:49 packer-provisioner-shell plugin: [WARN] Config file doesn't exist: /root/.packerconfig
2023/04/10 22:39:49 packer-provisioner-shell plugin: [INFO] Setting cache directory: /root/.cache/packer
2023/04/10 22:39:49 packer-provisioner-shell plugin: args: []string{"packer-provisioner-shell"}
2023/04/10 22:39:49 Received unix RPC address for /usr/local/bin/packer: addr is /tmp/packer-plugin2629057688
2023/04/10 22:39:49 packer-provisioner-shell plugin: Plugin address: unix /tmp/packer-plugin2629057688
2023/04/10 22:39:49 packer-provisioner-shell plugin: Waiting for connection...
2023/04/10 22:39:49 packer-provisioner-shell plugin: Serving a plugin connection...
2023/04/10 22:39:49 [TRACE] Starting internal plugin packer-provisioner-file
2023/04/10 22:39:49 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-provisioner-file"}
2023/04/10 22:39:49 Waiting for RPC address for: /usr/local/bin/packer
2023/04/10 22:39:49 packer-provisioner-file plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/04/10 22:39:49 packer-provisioner-file plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/04/10 22:39:49 packer-provisioner-file plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /root/.packerconfig
2023/04/10 22:39:49 packer-provisioner-file plugin: [WARN] Config file doesn't exist: /root/.packerconfig
2023/04/10 22:39:49 packer-provisioner-file plugin: [INFO] Setting cache directory: /root/.cache/packer
2023/04/10 22:39:49 packer-provisioner-file plugin: args: []string{"packer-provisioner-file"}
2023/04/10 22:39:49 Received unix RPC address for /usr/local/bin/packer: addr is /tmp/packer-plugin1849041675
2023/04/10 22:39:49 packer-provisioner-file plugin: Plugin address: unix /tmp/packer-plugin1849041675
2023/04/10 22:39:49 packer-provisioner-file plugin: Waiting for connection...
2023/04/10 22:39:49 packer-provisioner-file plugin: Serving a plugin connection...
2023/04/10 22:39:49 [TRACE] Starting internal plugin packer-provisioner-shell
2023/04/10 22:39:49 Starting plugin: /usr/local/bin/packer []string{"/usr/local/bin/packer", "plugin", "packer-provisioner-shell"}
2023/04/10 22:39:49 Waiting for RPC address for: /usr/local/bin/packer
2023/04/10 22:39:49 packer-provisioner-shell plugin: [INFO] Packer version: 1.8.6 [go1.18.9 linux amd64]
2023/04/10 22:39:49 packer-provisioner-shell plugin: [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/04/10 22:39:49 packer-provisioner-shell plugin: [INFO] PACKER_CONFIG env var set; attempting to open config file: /root/.packerconfig
2023/04/10 22:39:49 packer-provisioner-shell plugin: [WARN] Config file doesn't exist: /root/.packerconfig
2023/04/10 22:39:49 packer-provisioner-shell plugin: [INFO] Setting cache directory: /root/.cache/packer
2023/04/10 22:39:49 packer-provisioner-shell plugin: args: []string{"packer-provisioner-shell"}
2023/04/10 22:39:49 packer-provisioner-shell plugin: Plugin address: unix /tmp/packer-plugin2737240381
2023/04/10 22:39:49 packer-provisioner-shell plugin: Waiting for connection...
2023/04/10 22:39:49 Received unix RPC address for /usr/local/bin/packer: addr is /tmp/packer-plugin2737240381
2023/04/10 22:39:49 packer-provisioner-shell plugin: Serving a plugin connection...
2023/04/10 22:39:49 ui: [1;32mubuntu-server-focal.proxmox.ubuntu-server-focal: output will be in this color.[0m
2023/04/10 22:39:49 ui: 
2023/04/10 22:39:49 Build debug mode: false
2023/04/10 22:39:49 Force build: false
2023/04/10 22:39:49 On error: 
2023/04/10 22:39:49 Waiting on builds to complete...
2023/04/10 22:39:49 Starting build run: ubuntu-server-focal.proxmox.ubuntu-server-focal
2023/04/10 22:39:49 Running builder: 
2023/04/10 22:39:49 [INFO] (telemetry) Starting builder proxmox.ubuntu-server-focal
2023/04/10 22:39:49 packer-builder-proxmox plugin: using token auth
2023/04/10 22:39:49 packer-builder-proxmox plugin: No URLs were provided to Step Download. Continuing...
2023/04/10 22:39:49 ui: [1;32m==> ubuntu-server-focal.proxmox.ubuntu-server-focal: Creating VM[0m
2023/04/10 22:39:49 packer-builder-proxmox plugin: panic: runtime error: invalid memory address or nil pointer dereference
2023/04/10 22:39:49 packer-builder-proxmox plugin: [signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x2eedf37]
2023/04/10 22:39:49 packer-builder-proxmox plugin: 
2023/04/10 22:39:49 packer-builder-proxmox plugin: goroutine 121 [running]:
2023/04/10 22:39:49 packer-builder-proxmox plugin: github.com/Telmate/proxmox-api-go/proxmox.(*Client).CreateQemuVm(0xc000c6e0a0, {0xc000507d40, 0x11}, 0x51e4570?)
2023/04/10 22:39:49 packer-builder-proxmox plugin: 	github.com/Telmate/proxmox-api-go@v0.0.0-20221209160129-bb95d9a81e7a/proxmox/client.go:564 +0x297
2023/04/10 22:39:49 packer-builder-proxmox plugin: github.com/Telmate/proxmox-api-go/proxmox.ConfigQemu.CreateVm({0x0, {0xc000507cb0, 0x13}, {0x51e4570, 0x19}, {0x0, 0x0}, {0x0, 0x0}, 0xc001082210, ...}, ...)
2023/04/10 22:39:49 packer-builder-proxmox plugin: 	github.com/Telmate/proxmox-api-go@v0.0.0-20221209160129-bb95d9a81e7a/proxmox/config_qemu.go:213 +0x12af
2023/04/10 22:39:49 packer-builder-proxmox plugin: github.com/hashicorp/packer-plugin-proxmox/builder/proxmox/iso.(*isoVMCreator).Create(_, _, {0x0, {0xc000507cb0, 0x13}, {0x51e4570, 0x19}, {0x0, 0x0}, {0x0, ...}, ...}, ...)
2023/04/10 22:39:49 packer-builder-proxmox plugin: 	github.com/hashicorp/packer-plugin-proxmox@v1.1.1/builder/proxmox/iso/builder.go:64 +0xe5
2023/04/10 22:39:49 packer-builder-proxmox plugin: github.com/hashicorp/packer-plugin-proxmox/builder/proxmox/common.(*stepStartVM).Run(0xc000c76060, {0x5196368?, 0xc000c6a080?}, {0x59e2f28, 0xc000c62090})
2023/04/10 22:39:49 packer-builder-proxmox plugin: 	github.com/hashicorp/packer-plugin-proxmox@v1.1.1/builder/proxmox/common/step_start_vm.go:94 +0x9ac
2023/04/10 22:39:49 packer-builder-proxmox plugin: github.com/hashicorp/packer-plugin-sdk/multistep.(*BasicRunner).Run(0xc000c62150, {0x59e2128, 0xc000c6a080}, {0x59e2f28, 0xc000c62090})
2023/04/10 22:39:49 packer-builder-proxmox plugin: 	github.com/hashicorp/packer-plugin-sdk@v0.3.4/multistep/basic_runner.go:73 +0x2f3
2023/04/10 22:39:49 packer-builder-proxmox plugin: github.com/hashicorp/packer-plugin-proxmox/builder/proxmox/common.(*Builder).Run(0xc000c7c000, {0x59e2128, 0xc000c6a080}, {0x59e6fd0?, 0xc000c62060}, {0x59a5f20?, 0xc000c78000}, {0x59e2f28?, 0xc000c62090?})
2023/04/10 22:39:49 packer-builder-proxmox plugin: 	github.com/hashicorp/packer-plugin-proxmox@v1.1.1/builder/proxmox/common/builder.go:100 +0xde3
2023/04/10 22:39:49 packer-builder-proxmox plugin: github.com/hashicorp/packer-plugin-proxmox/builder/proxmox/iso.(*Builder).Run(0xc00046dc00, {0x59e2128, 0xc000c6a080}, {0x59e6fd0, 0xc000c62060}, {0x59a5f20, 0xc000c78000})
2023/04/10 22:39:49 packer-builder-proxmox plugin: 	github.com/hashicorp/packer-plugin-proxmox@v1.1.1/builder/proxmox/iso/builder.go:54 +0x3c5
2023/04/10 22:39:49 packer-builder-proxmox plugin: github.com/hashicorp/packer-plugin-sdk/rpc.(*BuilderServer).Run(0xc000138040, 0x856678?, 0xc00085665c)
2023/04/10 22:39:49 packer-builder-proxmox plugin: 	github.com/hashicorp/packer-plugin-sdk@v0.3.4/rpc/builder.go:120 +0x1e3
2023/04/10 22:39:49 packer-builder-proxmox plugin: reflect.Value.call({0xc000136cc0?, 0xc000134748?, 0x13?}, {0x518b294, 0x4}, {0xc00078fef8, 0x3, 0x3?})
2023/04/10 22:39:49 packer-builder-proxmox plugin: 	reflect/value.go:556 +0x845
2023/04/10 22:39:49 packer-builder-proxmox plugin: reflect.Value.Call({0xc000136cc0?, 0xc000134748?, 0x0?}, {0xc00104def8, 0x3, 0x3})
2023/04/10 22:39:49 packer-builder-proxmox plugin: 	reflect/value.go:339 +0xbf
2023/04/10 22:39:49 packer-builder-proxmox plugin: net/rpc.(*service).call(0xc000138080, 0x0?, 0x0?, 0xc000878440, 0xc0000caa80, 0x0?, {0x40aac20?, 0xc000856658?, 0x0?}, {0x3f7b3e0, ...}, ...)
2023/04/10 22:39:49 packer-builder-proxmox plugin: 	net/rpc/server.go:381 +0x226
2023/04/10 22:39:49 packer-builder-proxmox plugin: created by net/rpc.(*Server).ServeCodec
2023/04/10 22:39:49 packer-builder-proxmox plugin: 	net/rpc/server.go:478 +0x3fe
2023/04/10 22:39:49 [INFO] (telemetry) ending proxmox.ubuntu-server-focal
2023/04/10 22:39:49 ui error: [1;31mBuild 'ubuntu-server-focal.proxmox.ubuntu-server-focal' errored after 43 milliseconds 398 microseconds: unexpected EOF[0m
2023/04/10 22:39:49 ui: 
==> Wait completed after 43 milliseconds 455 microseconds
2023/04/10 22:39:49 machine readable: error-count []string{"1"}
2023/04/10 22:39:49 ui error: 
==> Some builds didn't complete successfully and had errors:
2023/04/10 22:39:49 machine readable: ubuntu-server-focal.proxmox.ubuntu-server-focal,error []string{"unexpected EOF"}
2023/04/10 22:39:49 ui error: --> ubuntu-server-focal.proxmox.ubuntu-server-focal: unexpected EOF
2023/04/10 22:39:49 ui: 
==> Builds finished but no artifacts were created.
2023/04/10 22:39:49 [INFO] (telemetry) Finalizing.
2023/04/10 22:39:49 /usr/local/bin/packer: plugin process exited
2023/04/10 22:39:49 waiting for all plugin processes to complete...
2023/04/10 22:39:49 /usr/local/bin/packer: plugin process exited
2023/04/10 22:39:49 /usr/local/bin/packer: plugin process exited
2023/04/10 22:39:49 /usr/local/bin/packer: plugin process exited

**Comment by nywilken** [comment]
@AlekKCode I believe this may have been fixed in the latest Proxmox plugin release. However, there is an issue in that the configuration being used refers to the builder as `proxmox` and not `proxmox-iso`. There is a PR to fix that issue #175 

Please let me know if making the following changes to the template resolve the issue for you.

```hcl
# Adding required_plugins to get the latest version of the Proxmox plugin
packer {
  required_plugins {
    proxmox = {
      version = ">= 1.1.2"
      source  = "github.com/hashicorp/proxmox"
    }
  }
}

#[redacted]

# Resource Definiation for the VM Template
source "proxmox-iso" "ubuntu-server-focal" {

#[redacted]
}

build {
 sources = ["source.proxmox-iso.ubuntu-server-focal"]
 
#[redacted]

}
```

**Comment by AlekKCode** [comment]
Thank you for replying, you fixed my issue. Much appreciated.
